src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,51 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
 
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size;
+
+    public MateStack() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void push(T value) {
+        resize();
+        array[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        isStackEmpty();
+        return array[size - 1];
     }
 
     public T pop() {
-        return null;
+        isStackEmpty();
+        T currentValue = array[size - 1];
+        array[size - 1] = null;
+        size--;
+        return currentValue;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        if (array.length == size) {
+            T[] newArray = (T[]) new Object[DEFAULT_CAPACITY * 3 / 2];
+            System.arraycopy(array, 0, newArray, 0, size);
+            array = newArray;
+        }
```

Comment:
I did the same and mentor told me that it should be two methods (check and grow).

Reply:
Thanks